src/tiled/commanddialog.cpp

```diff
@@ -63,6 +69,19 @@ void CommandDialog::closeEvent(QCloseEvent *event)
     CommandManager::instance()->updateActions();
 }
 
+void CommandDialog::setShortcut(const QKeySequence &keySequence)
+{
+    mUi->treeView->model()->setShortcut(mUi->treeView->currentIndex(), keySequence);
+}
+
+void CommandDialog::updateKeySequenceEdit(const QModelIndex &current, const QModelIndex &)
+{
+    if (current.row() < mUi->treeView->model()->rowCount(current))
```

Comment:
`rowCount(current)` will always return 0 unless current happens to be invalid, in which case it would be commandCount+1, so this check still seems wrong. I think you want:

```c++
if (current.row() < mUi->treeView->model()->rowCount() - 1)
```